### DISCUSSION TITLE
Docs/Naming: Standardize on "Promise" or "Thenable" in rule names

### DISCUSSION THREAD

**Discussion by JoshuaKGoldberg** [typescript-eslint/typescript-eslint#6012] [Technical Discussions]
### Before You File a Documentation Request Please Confirm You Have Done The Following...

- [X] I have looked for existing [open or closed documentation requests](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3Adocumentation) that match my proposal.
- [X] I have [read the FAQ](https://typescript-eslint.io/docs/linting/troubleshooting) and my problem is not listed.

### Suggested Changes

One rule directly refers to Thenables:

* [`@typescript-eslint/await-thenable`](https://typescript-eslint.io/rules/await-thenable)

Other rules refer to Promises:

* [`@typescript-eslint/no-floating-promises`](https://typescript-eslint.io/rules/no-floating-promises)
* [`@typescript-eslint/no-misused-promises`](https://typescript-eslint.io/rules/no-misused-promises)
* [`@typescript-eslint/promise-function-async`](https://typescript-eslint.io/rules/promise-function-async)

IMO it's confusing how only one rule refers to Thenables... even though that's technically the correct term. 

Thoughts: should we rename some rule(s) for clarity vs. discoverability? I haven't re-looked deeply into the rules, but if no-floating-promises & no-misused-promises actually work on all Thenables (and not just Promises) then maybe we should call them all thenables? Or maybe most users don't know what a Thenable is, and we should always say Promises?

### Affected URL(s)

https://typescript-eslint.io/rules/

**Comment by bradzacher** [comment]
IIRC the name `await-thenable` came from the tslint rule - which was written in a time back when using `bluebird` instead of promises was super duper common, so you needed to support "thenable" things instead of just "promises".

now-a-days yeah promises all the way.
It's probably a good idea to add this.

I wonder if we can also handle this from a documentation side of things - maybe adding keywords to rule metadata so that you can search for themes instead of relying solely on rule names?

**Reply by JoshuaKGoldberg** [reply]
I'll defer to @Josh-Cena on Docusaurus best practices 😄. But at the very least, making sure to include "thenable" at least once in all rules that relate to Promises is probably a good idea.

**Reply by JoshuaKGoldberg** [reply]
Ah, found the source of the rename: https://github.com/typescript-eslint/typescript-eslint/pull/192#issuecomment-464577047

@mysticatea if you have the time, I'd be curious to get your input on whether this rename should happen?
